server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -111,20 +115,64 @@ public KnnRetrieverBuilder(
         Float similarity
     ) {
         this.field = field;
-        this.queryVector = queryVector;
+        this.queryVector = () -> queryVector;
```

Comment:
Maybe we could also add a validation that only one of `queryVector` or `queryVectorBuilder` can be provided during parsing?

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/114366/commits/617dff766d6a9cfd19ab6dd5a13c080e66d30299